airstack-modules/modules/airstack/social/social.ts

```diff
@@ -1,998 +1,1180 @@
-    airBlock.save();
-    // creating air social user
-    const userId = createSocialUserEntityId(chainId, socialUserId);
-    // create air social user extras
-    let airSocialUserExtraIds = new Array<string>();
-    for (let i = 0; i < userExtras.length; i++) {
-      const extra = userExtras[i];
-      const airSocialUserExtraData = createAirExtra(
-        extra.name,
-        extra.value,
-        userId,
-      );
-      airSocialUserExtraIds.push(airSocialUserExtraData.id);
+    /**
+     * @dev this function tracks a air social profile ownership change transaction
+     * @param block ethereum block
+     * @param transactionHash transaction hash
+     * @param logOrCallIndex log or call index
+     * @param from erc721 token sender addres
+     * @param to erc721 token receiver address
+     * @param tokenId erc721 token id
+     * @param tokenAddress erc721 token address
+     * @param socialUserId social user id (eg: farcasterId)
+     */
+    export function trackSocialProfileOnwershipChangeTransaction(
+        block: ethereum.Block,
+        transactionHash: string,
+        logOrCallIndex: BigInt,
+        from: string,
+        to: string,
+        tokenId: string,
+        tokenAddress: string,
+        socialUserId: string
+    ): void {
+        const chainId = getChainId()
+        const airBlock = getOrCreateAirBlock(
+            chainId,
+            block.number,
+            block.hash.toHexString(),
+            block.timestamp
+        )
+        airBlock.save()
+        const airSocialProfile = getAirSocialProfile(chainId, tokenAddress, tokenId)
+        if (airSocialProfile == null) {
+            throw new Error("air social profile not found")
+        }
+        const airSocialUser = createAirSocialUser(
+            chainId,
+            airBlock,
+            socialUserId,
+            to,
+            new Array<string>()
+        )
+        airSocialProfile.user = airSocialUser.id
+        airSocialProfile.lastUpdatedAt = airBlock.id
+        airSocialProfile.save()
+        createAirSocialProfileOnwershipChangeTransaction(
+            chainId,
+            airBlock,
+            transactionHash,
+            logOrCallIndex,
+            from,
+            to,
+            tokenId,
+            tokenAddress,
+            airSocialProfile
+        )
     }
-    // create air social profile extras
-    let airSocialProfileExtraIds = new Array<string>();
-    const socialProfileEntityId = chainId.concat("-").concat(tokenAddress).concat("-").concat(tokenId);
-    for (let i = 0; i < profileExtras.length; i++) {
-      const extra = profileExtras[i];
-      const airSocialProfileExtraData = createAirExtra(
-        extra.name,
-        extra.value,
-        socialProfileEntityId,
-      );
-      airSocialProfileExtraIds.push(airSocialProfileExtraData.id);
+    /**
+     * @dev this function tracks when user sets defaultProfile
```

Comment:
please update this dev doc for track function.

Reply:
done